File: FE/WEB/app/models/media-category.js
import Model, { attr, hasMany, belongsTo } from '@ember-data/model';

export default class MediaCategoryModel extends Model {
  @hasMany('media-file', { async: true, inverse: 'typologyAreaRel' })
  typologyMediaFiles;
  @hasMany('media-file', { async: true, inverse: 'categoryRel' })
  categoryMediaFiles;
  @hasMany('media-file', { async: true, inverse: 'albumRel' }) albumMediaFiles;
  @belongsTo('tenant', { async: true, inverse: 'mediaCategory' }) tenant;

  @attr('string', { defaultValue: '' }) name;
  @attr('string', { defaultValue: '' }) code;
  @attr('string', { defaultValue: '' }) description;
  @attr('number', { defaultValue: 0 }) tenantId;
  @attr('boolean', { defaultValue: true }) erasable;
  @attr('boolean', { defaultValue: false }) copyInNewTenant;
  @attr('string', { defaultValue: '' }) parentMediaCategory;
  @attr('string', { defaultValue: '' }) type; // 'typology', 'category', 'album'
  @attr('number', { defaultValue: 0 }) order;
}

File: BE/Src/MIT.Fwk.Infrastructure/Entities/MediaCategory.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace MIT.Fwk.Infrastructure.Entities
{
    [Table("MediaCategories")]
    public class MediaCategory : Identifiable<string>
    {

        [Attr] public string Name { get; set; }
        [Attr] public string Code { get; set; }
        [Attr] public string Description { get; set; }
        [Attr] public int Order { get; set; }
        [Attr] public int TenantId { get; set; }
        [Attr] public bool Erasable { get; set; }
        [Attr] public string Type { get; set; }
        [Attr] public string ParentMediaCategory { get; set; }
        [Attr] public bool copyInNewTenant { get; set; }

        [HasOne]
        public virtual Tenant Tenant { get; set; }

        [HasMany]
        public virtual ICollection<MediaFile> TypologyMediaFiles { get; }

        [HasMany]
        public virtual ICollection<MediaFile> CategoryMediaFiles { get; }

        [HasMany]
        public virtual ICollection<MediaFile> AlbumMediaFiles { get; }
    }
}
